### PR TITLE
General Project Maintenance: dependency updates and test fixes

### PR DESCRIPTION
**Pull request by tgragnato**
This is a maintenance operation that will help keep our project up-to-date and robust.
It consists of dependency updates, test fixes, and integration upgrades.

**chore(deps): bump baseline, alpine, crypto**

- bump go version in github actions
- setup-go to v4 in github actions
- add test's cryptokeys to .gitignore
- some direct and indirect dependencies

**fix(test): remove the expired embedded x509 keypair**

This test is failing because the embedded cryptographic material has expired.

Instead of requiring the maintainer to regenerate the public/private keypair periodically, I am adding a function that, with each run of the unit tests, generates a new self-signed certificate with a validity period of 30 days.

**chore(deps): bump of the remaining dependencies**

- `aws-sdk-go`
- `gofiber/redirect`
- `google/uuid`
- `matryer/is`
- `spf13/cobra`
- indirect dependencies

**chore(deps): switch gofiber/template to gofiber/template/django/v3**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
         with:
-          go-version: "1.18.0-rc.1"
```
Comment (JellyBellyDev) [inline .github/workflows/test.yml:18]: try to update action to `actions/setup-go@v4`
https://github.com/golangci/golangci-lint-action?tab=readme-ov-file#how-to-use

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -26,7 +26,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
```
Comment (JellyBellyDev) [inline .github/workflows/test.yml:27]: also here try with `actions/setup-go@v4` ;)

**Comment by JellyBellyDev** [comment]
Good job @tgragnato !
Before merging this MR I would like to modify the package publishing system with a tag approach in order to always have a previous stable version.
WDYT?

/cc @streambinder

**Comment by tgragnato** [comment]
> Good job @tgragnato ! Before merging this MR I would like to modify the package publishing system with a tag approach in order to always have a previous stable version. WDYT?
> 
> /cc @streambinder

Thank you for the support!

In general, in projects that constitute critical infrastructure, it is always advisable to aim for the latest stable release so that the latest reliability and security updates are deployed. However, this is an operational concern; from a development standpoint, this project has now reached a certain level of stability, and there should be no drawbacks in the adoption of semantic versioning.

**Pull request merged**